File: Calculus.java
import java.util.*;
import java.io.*;

public class Calculus {
    public static void main(String[] args) throws FileNotFoundException{
        Scanner sc = new Scanner(new File("Calculus.in"));
        
        ArrayList<Integer> l = new ArrayList<>(Arrays.stream(sc.nextLine().split(" ")).map(Integer::parseInt).toList());
        int firstPow = l.size();

        for (int i = 0; i < firstPow; i++) {
            int currPow = firstPow - i;
            int newPow = currPow - 1;
            int currCoeff = l.get(i);
            int newCoeff = currCoeff * currPow;
            String operator = "";
            if (i != 0 && newCoeff >= 0) operator = "+";
            String xPow = "";
            if (newPow > 0) xPow= "x^" + newPow;
            System.out.print(operator + newCoeff + xPow);
        }
    }
}

File: Calculus.py
import sys
sys.stdin = open("Calculus.in", "r")

l: list[int] = list(map(int, input().split()))
firstPow: int = len(l)
for i in range(len(l)):
    currPow: int = firstPow - i
    newPow: int = currPow - 1
    currCoeff: int = l[i]
    newCoeff: int = currCoeff * currPow
    operator: str = ""
    if (i != 0 and newCoeff >= 0):
        operator = "+"
    xPow: str = ""
    if (newPow > 0):
        xPow = "x^" + str(newPow)
    print(operator, newCoeff, xPow, end = "", sep = "")